src/main/frontend/components/block.cljs

```diff
@@ -2118,18 +2118,15 @@
         data-refs-self (build-refs-data-value refs)
         edit-input-id (str "edit-block-" blocks-container-id "-" uuid)
         edit? (state/sub [:editor/editing? edit-input-id])]
-    [:div.ls-block.flex.flex-col.rounded-sm
+    [:div.ls-block
      (cond->
       {:id block-id
        :data-refs data-refs
        :data-refs-self data-refs-self
-       :style {:position "relative"}
        :class (str uuid
                    (when (and collapsed? has-child?) " collapsed")
                    (when pre-block? " pre-block"))
-       :blockid (str uuid)
-       :repo repo
-       :haschild (str has-child?)}
```

Comment:
Can I ask why do you remove `[haschild]` attribute here? the dev theme bullet threading depends on this value to fix an issue for multi-line blocks with children. ref https://github.com/pengx17/logseq-dev-theme/blob/fd8ad718f98c29d42b1097c5ea8624beb792e122/src/bullet_threading.scss#L81-L96

Reply:
@pengx17 The reason is that logseq's core code doesn't depend on this attribute anymore. But no worries, we can add it back :)

Reply:
I can add some helper attributes in another PR. Some of my plugins may need them in some cases, e.g., cmd+clicking an item in the search result to add a new tab.

Reply:
@pengx17 Adding more attributes will make the dom too heavy. Is a database query working for this use case?

Reply:
I think adding these data attributes should not have any real performance impacts though. They should be fairly simple and no other side-effects.
![image](https://user-images.githubusercontent.com/584378/141100158-d87762f0-275b-4e06-8015-d25ef6fbb44c.png)

Reply:
We might use the same code on the backend for HTML output, would like to know more about your use cases.

Reply:
For the tabs plugin, I will register a `mousedown` event handler in the main window and intercept clicks that are bubbled from page links etc, so that I can put them into the tabs array, persist and render them.
The missing part for the page links in the search item is the `original-name` or block-ref of the pages. I think the `textContent` in the menu item is not a very reliable data source for my purpose.

I am not sure what is the impact of adding them into the HTML output, whether or not in the backend. It should be fairly a simple operation.